lib/workers/repository/onboarding/branch/index.ts

```diff
@@ -47,16 +49,15 @@ export async function checkOnboardingBranch(
   setGitAuthor(config.gitAuthor);
   const onboardingPr = await getOnboardingPr(config);
   if (onboardingPr) {
+    logger.debug('Onboarding PR already exists');
     if (config.onboardingRebaseCheckbox) {
       handleOnboardingManualRebase(onboardingPr);
     }
-    logger.debug('Onboarding PR already exists');
-    const { rawConfigHash } = onboardingPr.bodyStruct ?? {};
-    const commit = await rebaseOnboardingBranch(config, rawConfigHash);
-    if (commit) {
-      logger.info(
-        { branch: config.onboardingBranch, commit, onboarding: true },
-        'Branch updated'
+    if (await scm.isBranchModified(config.onboardingBranch!)) {
+      invalidateExtractCache(config.baseBranch!);
```

Comment:
scm.isBranchModified returns whether a user has made a commit to the branch. I don't see why that automatically should trigger invalidating the cache. What if that commit they made was done months ago? We should be able to keep a cache on that

Reply:
Done. Working on tests rn but will commit once you confirm them.
Have used this logic-flow:
- Cache is only set once per run (at the end)
- Everywhere else cache is only called if its outdated we fall back to git else we return the cached result

Setting cache only once per run, prevents confusion which coould be caused due to multiple cache updates. WDYT??